### ISSUE TITLE
Url fragment is lost due to redirection

### ISSUE THREAD

**Issue by runephilosof-abtion** [rubygems/bundler-site#1347]
On https://bundler.io/guides/using_bundler_in_applications.html#gemfilelock

https://github.com/rubygems/bundler-site/blob/4121bcabc4f88c6f413c93af14c87e939012dc32/source/localizable/guides/using_bundler_in_applications.en.html.md?plain=1#L224

The link to `/man/gemfile.5.html#PLATFORMS` will be redirected to `/v2.5/man/gemfile.5.html`, by the backend https://github.com/rubygems/bundler-site/blob/4121bcabc4f88c6f413c93af14c87e939012dc32/config.rb#L185

So the fragment `#PLATFORMS` will never have an effect.

Could it link to the current version instead?

**Comment by deivid-rodriguez** [comment]
Indeed. I'm unsure how to fix this. We could make this guide use a haml template instead of markdown and dinamically link to the latest version, or hardcode the latest version in the target URL, and make sure we update it every time we release a new minor version of Bundler. None of them feel great :(

**Comment by runephilosof-abtion** [comment]
I just realized that browsers of course persist the `#PLATFORMS` part, when they are redirected.
However, it seems the backend is not redirecting with a http status 302 (or 307), but instead responding with 200 OK and this body.
```
              <html>
                <head>
                  <link rel="canonical" href="/v2.6/man/gemfile.5.html" />
                  <meta http-equiv=refresh content="0; url=/v2.6/man/gemfile.5.html" />
                  <meta name="robots" content="noindex,follow" />
                  <meta http-equiv="cache-control" content="no-cache" />
                </head>
                <body>
                </body>
              </html>
```

If the backend would instead use a redirection status and location header, it should work.

**Comment by runephilosof-abtion** [comment]
I can't find a way to get github pages to send a redirection response.

**Comment by deivid-rodriguez** [comment]
I think middleman always uses client side redirects. This seems bad for SEO and I think it explains the issues in https://github.com/rubygems/bundler-site/issues/1333.

I think we can fix both this and https://github.com/rubygems/bundler-site/issues/1333 by not using middleman `redirect` and instead creating symlinks between `/man/` and `v2.6/man` (updated every time we bump the major version). We will serve duplicated content, but our canonical `meta` tags should keep Google happy I believe.

**Comment by deivid-rodriguez** [comment]
I tried implementing this approach in #1424.
